lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -71,81 +71,131 @@ void SetMessageManagerHandler::SetAmfSessionContext(
   auto& request_cpy = *request;
   // Print the message from AMF
   PrintGrpcMessage(static_cast<const google::protobuf::Message&>(request_cpy));
-
   // Requested message from AMF to release the session
+
   m5g_enforcer_->get_event_base().runInEventBaseThread([this, response_callback,
                                                         request_cpy]() {
     // extract values from proto
     auto imsi       = request_cpy.common_context().sid().id();
     std::string dnn = request_cpy.common_context().apn();
+    // Fetch PDU session ID from rat_specific_context and
+    // pdu_id is unique to IMSI
+    auto pdu_id = request_cpy.rat_specific_context()

```

Comment:
If it's not too difficult to do so, let's use explicit type rather than auto. (It makes reading code easier at a later time. :) )

Reply:
Replaced auto keyword with explicit type (unit32_t for pdu_id)